account_mass_reconcile/models/base_advanced_reconciliation.py

```diff
@@ -214,16 +215,60 @@ def _skip_line(self, move_line):
         """
         return False
 
+    def _rec_group(self, reconcile_groups, lines_by_id):
+        reconciled_ids = []
+        for group_count, reconcile_group_ids \
+                in enumerate(reconcile_groups, start=1):
+
+            _logger.debug("Reconciling group %d/%d with ids %s",
+                          group_count, len(reconcile_groups),
+                          reconcile_group_ids)
+            group_lines = [lines_by_id[lid]
+                           for lid in reconcile_group_ids]
+            reconciled, full = self._reconcile_lines(group_lines,
+                                                     allow_partial=True)
+            if reconciled and full:
+                reconciled_ids += reconcile_group_ids
+        return reconciled_ids
+
+    def _rec_group_by_chunk(self, reconcile_groups, lines_by_id, chunk_size):
+        """Commit after each chunk
+
+        :param dict reconcile_grous: all groups to reconcile, will be splitted
+        by chunk
+        :param list lines_by_id: list of dict of move lines values,
+          the move lines we want to search for
+        :return: list of reconciled lines
+        """
+        reconciled_ids = []
+
+        _logger.info("Reconciling by chunk of %d", chunk_size)
+
+        # Copy and commit current transient model before creating a new cursor
+        # This is required to avoid CacheMiss when using data from `self`
+        # which is created during current transaction.
+        with odoo.api.Environment.manage():
+            with odoo.registry(self.env.cr.dbname).cursor() as new_cr:
+                new_env = api.Environment(new_cr, self.env.uid, self.env.context)
+                self_env = self.with_env(new_env)
+                rec = self_env.create(self.copy_data())
```

Comment:
A bit of an ugly solution, but if it works and avoids the bare `cr.commit`, it's still less ugly than the actual code. :+1: 

Maybe add a TODO to refactor this so we don't have to create and persist actual wizard but use the result of `copy_data` with a new instead of a `browse` below. :question:

Reply:
The wizard will be garbage collected anyway. I doubt this wizard will be called enough time per day even with a cron to cause an issue in term of DB size. For version 12.0 we won't invest time on refactoring the whole mass reconcile process, it would mean messing with the method signatures to use staticmethods instead. I would prefer to keep the code stable.